### PR TITLE
Clean up code-cleanup.sh and make change more explicit

### PR DESCRIPTION
**Pull request by wesbland**
This tweaks the settings of the intent script in code-cleanup.sh and
expands the K&R setting to include all of what it means under the hood.
This should help slightly when running the cleanup script on new files.

### CODE REVIEW

**Comment by raffenet** [comment]
What the tweaks here vs. just expanding the K&R flag?

**Comment by wesbland** [comment]
No. It's more than that. I'd have to go parse the man page to get the expansion.

**Comment by raffenet** [comment]
Maybe make it 2 commits then?

**Comment by wesbland** [comment]
That should be clearer now as two commits. Good suggestion.

File: maint/code-cleanup.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #! /bin/bash
 
-if test ! -z "`which gindent`" ; then
+if hash gindent 2>/dev/null; then
```
Comment (raffenet) [inline maint/code-cleanup.sh]: Probably way more hairsplitting that needed on this PR, but why this change? If needed, should be 3 commits total, IMO.

Comment (wesbland) [inline maint/code-cleanup.sh]: I don't remember the specifics now, but somewhere it wasn't detecting indent vs. gindent correctly. Making this change made it more reliable. We can drop this if you care. I don't think it matters to me.

Comment (raffenet) [inline maint/code-cleanup.sh]: K, let's just drop it for now.

Comment (wesbland) [inline maint/code-cleanup.sh]: Done.

File: maint/code-cleanup.sh [hunk 2]
```diff
@@ -10,10 +10,56 @@ indent_code()
 {
     file=$1
 
-    $indent --k-and-r-style --line-length100 --else-endif-column1 --start-left-side-of-comments \
-	--break-after-boolean-operator --dont-cuddle-else --dont-format-comments \
-	--comment-indentation1 --indent-level4 --no-tabs \
-	${file}
+    $indent \
+        \ # Expansion of Kernighan & Ritchie style
```
Comment (raffenet) [inline maint/code-cleanup.sh]: So my `bash` doesn't like these inline style comments and won't run the script. Stackoverflow recommends using backticks...

Comment (raffenet) [inline maint/code-cleanup.sh]: https://stackoverflow.com/questions/9522631/how-to-put-line-comment-for-a-multi-line-command#12797512

**Comment by wesbland** [comment]
My latest changes should fix the issues you raised @raffenet. I also put the change back in for `indent` vs. `gindent`. The problem that was coming up was that a bunch of extra debug output shows up if you have `indent` instead of `gindent`.
